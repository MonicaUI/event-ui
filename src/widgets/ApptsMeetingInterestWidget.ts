import { connect } from 'react-redux';
import withMappedWidgetConfig from 'nucleus-widgets/lib/withMappedWidgetConfig';
import { setAppointmentsMeetingInterest } from '../redux/registrationForm/regCart/actions';
import { getEventRegistrationId } from '../redux/selectors/currentRegistrant';
import { getRegCart } from '../redux/selectors/shared';
import { getExhibitorCustomFields, loadExhibitorsList } from 'event-widgets/redux/modules/appointments';
import { getIn } from 'icepick';

import ApptsMeetingInterestWidget from 'event-widgets/lib/ApptsMeetingInterest/ApptsMeetingInterestWidget';

/* Returns appointments meeting interest from reg cart */
export const getApptsMeetingInterest = (state: $TSFixMe, eventRegistrationId: $TSFixMe): $TSFixMe => {
  const regCart = getRegCart(state);
  return getIn(regCart, ['eventRegistrations', eventRegistrationId, 'appointmentsMeetingInterest']);
};

export default withMappedWidgetConfig(
  connect(
    (state: $TSFixMe, props: $TSFixMe) => {
      const eventRegistrationId = getEventRegistrationId(state);
      return {
        config: props.config,
        meetingInterest: getApptsMeetingInterest(state, eventRegistrationId),
        eventRegistrationId,
        appointmentEvent: state.appointments.appointmentEvent,
        exhibitors: state.appointments.exhibitors,
        exhibitorCustomFieldsMap: state.appointments.exhibitorCustomFieldsMap,
        countries: state.countries.countries,
        isGuest: true
      };
    },
    {
      setAppointmentsMeetingInterest,
      getExhibitorCustomFields,
      loadExhibitorsList
    },
    (stateProps: $TSFixMe, dispatchProps: $TSFixMe, ownProps: $TSFixMe) => {
      return {
        ...ownProps,
        ...stateProps,
        setAppointmentsMeetingInterest: dispatchProps.setAppointmentsMeetingInterest,
        getExhibitorCustomFields: customFieldIds => dispatchProps.getExhibitorCustomFields(true, customFieldIds),
        loadExhibitorsList: customFieldIds => dispatchProps.loadExhibitorsList(true, customFieldIds)
      };
    }
  )(ApptsMeetingInterestWidget)
);
