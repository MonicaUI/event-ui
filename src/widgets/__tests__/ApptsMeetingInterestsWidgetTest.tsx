import React from 'react';
import ApptsMeetingInterestWidget from '../ApptsMeetingInterestWidget';
import { mount } from 'enzyme';

const loadMetaDataFunction = jest.fn(() => {
  return {};
});

function getState() {
  return {
    appointments: {
      appointmentEvent: { isValid: true },
      exhibitors: [{ id: 'testId', name: 'Test Exhibitor' }]
    },
    countries: {
      countries: {}
    },
    text: { translate: x => x },
    registrationForm: {
      warnings: {},
      regCart: {
        eventRegistrations: {
          REG1: {
            id: 'REG1',
            attendeeType: 'ATTENDEE',
            attendee: {
              personalInformation: {
                emailAddress: 'tony@stark.com'
              }
            },
            confirmationNumber: 'CONFIRMNUM'
          }
        }
      }
    },
    defaultUserSession: {
      isPlanner: false
    },
    plannerRegSettings: {
      exitUrl: 'https://www.google.com/'
    },
    event: {
      registrationTypes: {
        '00000000-0000-0000-0000-000000000000': {
          id: '00000000-0000-0000-0000-000000000000'
        }
      }
    },
    website: {
      pluginData: {
        registrationProcessNavigation: {
          registrationPaths: {
            regPathId: {
              id: 'regPathId',
              pageIds: ['pageId'],
              cancellationPageIds: ['cancellationPageId'],
              confirmationPageId: 'confirmationPageId'
            }
          }
        }
      },
      siteInfo: {
        sharedConfigs: {}
      }
    },
    widgetFactory: {
      loadMetaData: loadMetaDataFunction
    },
    appData: {
      registrationSettings: {
        registrationPaths: {
          regPathId: {
            id: 'regPathId',
            isDefault: true
          }
        }
      }
    }
  };
}

const subscribe = () => {};
const defaultProps = {
  classes: {},
  style: {
    palette: {},
    elements: { link: {} },
    itemNameContainer: { customSettings: { spacing: { padding: { paddingTop: 10 } } } }
  },
  translate: c => c,
  store: { dispatch, getState, subscribe }
};
const defaultPropsV1 = {
  ...defaultProps,
  config: {
    appData: {
      companyList: ['Company A', 'Company B']
    }
  }
};
const defaultPropsV2 = {
  ...defaultProps,
  config: { appData: { version: 2 } }
};

async function dispatch(action) {
  if (typeof action === 'function') {
    await action(dispatch, getState);
  }
}

describe('ApptsMeetingInterestWidget', () => {
  test('should match snapshot - old version', () => {
    const widget = mount(<ApptsMeetingInterestWidget {...defaultPropsV1} />);
    expect(widget.props()).toMatchSnapshot();
  });

  test('should match snapshot - new version', () => {
    const widget = mount(<ApptsMeetingInterestWidget {...defaultPropsV2} />);
    expect(widget.props()).toMatchSnapshot();
  });
});
