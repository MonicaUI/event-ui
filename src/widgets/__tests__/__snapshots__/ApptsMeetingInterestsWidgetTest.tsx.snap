// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApptsMeetingInterestWidget should match snapshot - new version 1`] = `
Object {
  "classes": Object {},
  "config": Object {
    "appData": Object {
      "version": 2,
    },
  },
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "subscribe": [Function],
  },
  "style": Object {
    "elements": Object {
      "link": Object {},
    },
    "itemNameContainer": Object {
      "customSettings": Object {
        "spacing": Object {
          "padding": Object {
            "paddingTop": 10,
          },
        },
      },
    },
    "palette": Object {},
  },
  "translate": [Function],
}
`;

exports[`ApptsMeetingInterestWidget should match snapshot - old version 1`] = `
Object {
  "classes": Object {},
  "config": Object {
    "appData": Object {
      "companyList": Array [
        "Company A",
        "Company B",
      ],
    },
  },
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "subscribe": [Function],
  },
  "style": Object {
    "elements": Object {
      "link": Object {},
    },
    "itemNameContainer": Object {
      "customSettings": Object {
        "spacing": Object {
          "padding": Object {
            "paddingTop": 10,
          },
        },
      },
    },
    "palette": Object {},
  },
  "translate": [Function],
}
`;
